docs/specification/appendix.md

```diff
@@ -118,4 +124,67 @@ The `ros2` profile describes how to create MCAP files for [ROS 2](https://docs.r
 
 #### Schema
 
-- `encoding`: MUST contain `ros2msg`
+- `encoding`: MUST contain either `ros2msg` or `ros2idl`
+
+## `gendeps`-style message definition delimiting
+
+For ROS1 and ROS2 message types, the message definition is stored in the [Schema record](./README.md#schema-op0x03) along with the definitions of any non-primitive field types. These definitions are concatenated together using a simple delimiting scheme:
+
+- The full content of the top-level message definition file is stored first.
+- Each dependent definition is preceded by:
```

Comment:
One thing I've always been annoyed with in the ROS1 "combined message" format is that the first message does not have this "separator" information. I think that we should have even the first message (i.e. the entry message) be preceded by a separator with its file information.

An additional benefit is that when you start reading the file you immediately see that its not a regular .idl message but instead a special separate format.

Reply:
This comment is for the new `ros2idl` format - not for the `ros2msg` format.

Reply:
Seems a little hard to justify diverging from the others just for this schema encoding - if we're willing to diverge at all, why not just do something like a JSON list of objects?

Reply:
> why not just do something like a JSON list of objects?

Now I need JSON library to read the schemas. It's nice to avoid that since these are text and what I am proposing is to add the leading separator so we have the same file and name information for that. In ROS1 there is no leading separator so you don't actually know the _datatype_ of the root message.

Reply:
you do know, it's in the schema `name`.

Reply:
From a function to parse this joined definitions its nice to be able to do `parseToAst(joinedText)` rather than `parseToAst(joinedText, "root_msg_datatype")` the latter we need to do today for ros1 messages because the root message metadata is not present.